articles/test/test_views.py

```diff
@@ -0,0 +1,250 @@
+import json
+from typing import Any
+from unittest.mock import patch
+
+from django.contrib.auth import get_user_model
+from django.test.client import BOUNDARY, MULTIPART_CONTENT, encode_multipart
+from faker import Faker
+from rest_framework import status
+from rest_framework.reverse import reverse
+from rest_framework.test import APIClient, APITestCase
+
+from articles.models import Article
+from articles.test.mocks import sample_image
+
+fake = Faker()
+User = get_user_model()
+
+
+class TestArticleViews(APITestCase):
+    def setUp(self) -> None:
+        self.client = APIClient()
+        self.password = fake.password()
+        self.user = User.objects.create_user(
+            username=fake.name(), email=fake.email(), password=self.password
+        )
+        self.article = Article.objects.create(
+            title=fake.name(),
+            description=fake.text(),
+            body=fake.text(),
+            image=sample_image(),
+            is_hidden=False,
+            favorited=False,
+            favoritesCount=0,
+            author=self.user,
+        )
+
+    @property
+    def bearer_token(self) -> Any:
+        login_url = reverse("login")
+        response = self.client.post(
+            login_url,
+            data={"email": self.user.email, "password": self.password},
+            format="json",
+        )
+        token = json.loads(response.content).get("access")  # type: ignore[attr-defined]
+        return {"HTTP_AUTHORIZATION": f"Bearer {token}"}
+
+    @patch(
+        "cloudinary.uploader.upload_resource", return_value=fake.image_url()
+    )
+    def test_create_article(self, upload_resource: None) -> None:
+        data = {
+            "title": fake.texts(nb_texts=2),
+            "description": fake.paragraph(nb_sentences=3),
+            "body": fake.paragraph(nb_sentences=20),
+            "image": sample_image(),
+            "is_hidden": False,
+            "tags": f'["{fake.word()}", "{fake.word()}"]',
+            "favorited": False,
+            "favoritesCount": 0,
+        }
+
+        response = self.client.post(
+            reverse("articles"),
+            data=encode_multipart(data=data, boundary=BOUNDARY),
+            content_type=MULTIPART_CONTENT,
+            enctype="multipart/form-data",
+            **self.bearer_token,
+        )
+        self.assertTrue(upload_resource.called)  # type: ignore[attr-defined]
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Article.objects.count(), 2)
```

Comment:
don't use hard coded values, check the count before the requeest then check it increased by 1, trust me hard coded values will bring issues in the future, change everywhere you've done that

Reply:
ok